accounting/acct.proto

```diff
@@ -0,0 +1,147 @@
+// Accounting for gnXi operations.
+//
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// A protobuffer definition used for logging events reported to gNxI server.
+// This file assumes the following nomenclature:
+//
+//   Device >---connects---> gNxI accounting Server
+//
+// Device is a router, switch, PC, ... which connects to gNxI accounting server
+// to log accounting records.
+//
+// A gNxI accounting Server is a server which receives requests from devices and
+// logs them in the format specified using messages defined in this file.
+//
+// The current version covers event recording of shell/vendor-CLI and gRPC
+// based connections.
+
+syntax = "proto3";
+
+package gnsi.accounting;
+
+option go_package = "github.com/openconfig/gnsi/accounting";
+
+// Session identifier that identifies the context in which service
+// was executed.
+message SessionInfo {
+  // address of the device sending the event.
+  string device_address = 1;
+
+  // rest of the information for 5 tuple of the connection
+  uint32 layer4_proto = 2;
+  uint32 local_port = 3;
+  string remote_address = 4;
+  uint32 remote_port = 5;
+
+  // For multiplexing multiple channels over single 5 tuple
+  // - multiple CLI connections over single SSH
+  string channel_id = 6;
+}
+
+// Details of authentication - for gRPC or shell/vendor-CLI.
+message AuthenDetail {
+  // Identity string is used to identify the user that executed the
+  // command. For instance, it could be spiffe-id in the case of gRPC or
+  // unix-style user-name in the case of shell/vendor-CLI.
+  string identity = 1;
+
+  // Privilege level configured on the device.
+  uint32 privilege_level = 2;
+
+  // authentication status
+  enum AuthenStatus {
+    AUTHEN_STATUS_UNKNOWN = 0;
+    AUTHEN_STATUS_PERMIT = 1;
+    AUTHEN_STATUS_DENY = 2;
+  }
+  AuthenStatus status = 3;
+
+  // In case of STATUS_DENY, cause for the deny
+  string deny_cause = 4;
```

Comment:
Is there any use case within which we want this to be an enumerated value vs. a free-text string?

Reply:
That makes sense to me -- I think I was thinking that it might be possible to have more structure such as being able to determine that "password mismatch" occurred without needing to string match. But in practice, I'm not sure that there's actually much use case whereby that would add significant value. I think you need the two fields that you've already defined, and in the future could augment this if we thought that there was some value there, but this SGTM as a first pass.